gemini/evaluation/evaluate_multimodal_task_image.ipynb

```diff
@@ -0,0 +1,758 @@
+    "id": "GTL_YzF9OvFq"
+   },
+   "outputs": [],
+   "source": [
+    "# Use the environment variable if the user doesn't provide Project ID.\n",
+    "import os\n",
+    "\n",
+    "import vertexai\n",
+    "\n",
+    "PROJECT_ID = \"[your-project-id]\"  # @param {type: \"string\", placeholder: \"[your-project-id]\", isTemplate: true}\n",
+    "\n",
+    "if not PROJECT_ID or PROJECT_ID == \"[your-project-id]\":\n",
+    "    PROJECT_ID = str(os.environ.get(\"GOOGLE_CLOUD_PROJECT\"))\n",
+    "\n",
+    "LOCATION = os.environ.get(\"GOOGLE_CLOUD_REGION\", \"us-central1\")\n",
+    "\n",
+    "vertexai.init(project=PROJECT_ID, location=LOCATION)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "EdvJRUWRNGHE"
+   },
+   "source": [
+    "## Evaluate your multimodal use case with custom metric\n",
+    "\n",
+    "Imagine you have a customer insurance app which provides several services including live incident support using an LLM. Given a conversation about a car accident and an image of the damaged car, you want to evaluate evaluate the coherence of the generated responses.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "5303c05f7aa6"
+   },
+   "source": [
+    "### Import libraries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "6fc324893334"
+   },
+   "outputs": [],
+   "source": [
+    "# General\n",
+    "from IPython.display import HTML, Markdown, display\n",
+    "from vertexai.evaluation import CustomMetric, EvalTask\n",
+    "from vertexai.generative_models import (GenerationConfig, GenerativeModel,\n",
+    "                                        HarmBlockThreshold, HarmCategory, Part)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "tfQ7sPtOjZOw"
+   },
+   "source": [
+    "### Library settings"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "cellView": "form",
+    "id": "RjWUgU1TjZOw"
+   },
+   "outputs": [],
+   "source": [
+    "# @title\n",
+    "import json\n",
+    "import logging\n",
+    "import warnings\n",
+    "\n",
+    "import pandas as pd\n",
+    "\n",
+    "logging.getLogger(\"urllib3.connectionpool\").setLevel(logging.ERROR)\n",
+    "warnings.filterwarnings(\"ignore\")\n",
+    "\n",
+    "# pd.set_option('display.max_colwidth', None)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "QfaFPmm7gC2V"
+   },
+   "source": [
+    "### Helpers"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "cellView": "form",
+    "id": "2kyy7mGhgEj9"
+   },
+   "outputs": [],
+   "source": [
+    "# @title\n",
+    "def print_doc(function):\n",
+    "    print(f\"{function.__name__}:\\n{inspect.getdoc(function)}\\n\")\n",
+    "\n",
+    "\n",
+    "def display_eval_result(eval_result, title=None, metrics=None):\n",
+    "    \"\"\"Display the evaluation results.\"\"\"\n",
```

Comment:
Can you add types for the parameters and return values?

Reply:
done